chall06/mzomeno-.c

```diff
@@ -0,0 +1,173 @@
+static int get_line_length(char *board)
+{
+  int line_length;
+
+  line_length = 0;
+  while (*board != '\n' && *board != '\0')
+    {
+      if (*board != '.' && *board != 'K' &&
+	  *board != 'P' && *board != 'R' &&
+	  *board != 'B' && *board != 'Q')
+	return (0);
+      line_length++;
+      board++;
+    }
+  return (line_length);
+}
+
+static int check_board(char *board, int line_length)
+{
+  int line_number;
+
+  if (line_length == 0 || line_length > 9)
+    return (0);
+  line_number = 1;
+  while (*board != '\0')
+    {
+      if (line_length != get_line_length(board + (line_length * (line_number + 1))))
+	return (1);
+      line_number++;
+      board++;
+    }
+  if (line_number != line_length)
+    return (0);
+  return (line_number);
+}
+
+static int check_king(char *board)
+{
+  int iterator;
+  int king_position;
+
+  king_position = -1;
+  iterator = 0;
+  while (board[iterator] != '\0')
+    {
+      if (board[iterator] == 'K')
+	{
+	  if (king_position == -1)
+	    king_position = iterator;
+	  else
+	    return (-1);
+	}
+      iterator++;
+    }
+  return (king_position);
+  /*
+   * Note that the function will return -1 if there is no king,
+   * as the king_position variable’s initial value remains untouched
+   */
+}
+
+static int pawn_capture(int line_length, int position, int king_position)
+{
+  if (king_position == (position - line_length - 2) ||
+      king_position == (position - line_length + 2) ||
+      king_position == (position + line_length - 2) ||
+      king_position == (position + line_length + 2))
+    return (1);
+  else
+    return (0);
+}
+
+static int bishop_capture(int line_length, int line_number, int position, int king_position)
+{
+  int n;
+
+  n = 1;
+  while (n < (line_number - (position - line_length)))
+    {
+      if (king_position == (position + n * (line_length + 1) + n) ||
+	  king_position == (position + n * (line_length + 1) - n))
+	return (1);
+      n++;
+    }
+  n = 1;
+  while (n < (position / line_length))
+    {
+      if (king_position == (position - n * (line_length + 1) + n) ||
+	  king_position == (position - n * (line_length + 1) - n))
+	return (1);
+      n++;
+    }
+  return (0);
+}
+
+static int rook_capture(int line_length, int line_number,
+			int position, int king_position)
+{
+  int n;
+
+  n = 1;
+  while (n < (line_number - (position - line_length)))
+    {
+      if (king_position == (position + n * (line_length + 1)) ||
```

Comment:
There's probably an error in this condition and you never find the King ('`K`') in the same line as a Rook ('`R`'). Probably should have stored the line the king is in for an easier Rook ('`R`') check.